File: 01-Numbers/py/variables.py
if __name__ =="__main__":
    a = 42
    print(a)
    #shadowing the value
    a = "forty two"
    print(a)

    # declare, init
    opt = None
    a = 15
    print("x, opt", a, opt)
    opt = a * 5
    print("x, opt", a, opt)

    #python variabkles are mutable
    b = 0
    print(f"before b == {b}")
    b = (b + 3) * opt
    print(f"after b == {b}")

    FOO = 42  # only const by convention
    a *= FOO

File: 01-Numbers/rs/src/variables.rs
pub fn using() {
    let a = 42;
    println!("a: {}", a);
    //shadowing the value
    let a = "forty two".to_string();
    println!("a: {}", a);

    // declare, initialize
    let something: Option<u32> = None;
    let x = 5;
    println!("x, something: {}, {:?}", x, something);
    let something = Some(x * 5);
    println!("x, something: {}, {:?}", x, something);

    // Mutability
    let mut y = 0;
    y = y * 2 + x;
    dbg!(y);

    const FOO: i32 = 42;
    let y = y * FOO;
    dbg!(y);
}
